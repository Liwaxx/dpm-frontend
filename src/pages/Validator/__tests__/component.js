import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import renderer from 'react-test-renderer';
import configureStore, { history } from '../../../store/configureStore';
import Component from '../component';

jest.mock('../../../components/elements/AppBar');
jest.mock('../../../components/elements/ValidatorCard');
jest.mock('../../../components/elements/Text');

let actions, data;

describe('Component', () => {
  beforeEach(() => {
    actions = {
      fetchValidator: jest.fn(),
    };
    data = {
      approved: {
        today: [],
        yesterday: [],
        oneWeek: []
      },
      rejected: {
        today: [],
        yesterday: [],
        oneWeek: []
      },
      waiting: {
        today: [],
        yesterday: [],
        oneWeek: []
      },
    };
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Component actions={actions} classes={{}} data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders loading state', () => {
    const tree = renderer
      .create(<Component actions={actions} classes={{}} data={data} isLoading />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with data', () => {
    data = {
      approved: {
        today: ['test'],
        yesterday: ['test'],
        oneWeek: [
          {
            data: ['test'],
            id: '2018-11-11',
          }
        ]
      },
      rejected: {
        today: ['test'],
        yesterday: ['test'],
        oneWeek: [
          {
            data: ['test'],
            id: '2018-11-11',
          }
        ]
      },
      waiting: {
        today: ['test'],
        yesterday: ['test'],
        oneWeek: [
          {
            data: ['test'],
            id: '2018-11-11',
          }
        ]
      },
    };
    const tree = renderer
      .create(
        <Provider store={configureStore()}>
          <Router history={history}>
            <Component
              actions={actions}
              classes={{}}
              data={data}
            />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});