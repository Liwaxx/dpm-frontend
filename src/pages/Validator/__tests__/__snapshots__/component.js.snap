// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<section>
  <header />
  <div>
    <div>
      <div />
      <div />
    </div>
    <div>
      <div />
      <div />
    </div>
    <div>
      <div />
      <div />
    </div>
  </div>
</section>
`;

exports[`Component renders loading state 1`] = `
<section>
  <header />
  <div>
    <div>
      <div />
      <div>
        <div
          className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg-6"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <div />
      <div>
        <div
          className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg-6"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <div />
      <div>
        <div
          className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg-6"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Component renders with data 1`] = `
<section>
  <header />
  <div>
    <div>
      <div />
      <div />
    </div>
    <div>
      <div />
      <div />
    </div>
    <div>
      <div />
      <div />
    </div>
  </div>
</section>
`;
